src/pages/strategies/forms/indicatorForm.js

```diff
@@ -0,0 +1,206 @@
+import React, { useState, useEffect, useContext } from 'react'
+import { Box, Button } from '@mui/material'
+import { useFormik } from 'formik'
+import * as yup from 'yup'
+import { RequestsContext } from 'src/providers/RequestsContext'
+import toast from 'react-hot-toast'
+import { DocumentReleaseRepository } from 'src/repositories/DocumentReleaseRepository'
+import { DataGrid } from 'src/components/Shared/DataGrid'
+import FormShell from 'src/components/Shared/FormShell'
+
+import { ResourceIds } from 'src/resources/ResourceIds'
+
+const IndicatorForm = ({
+  labels,
+  strategiesFormik,
+  expanded,
+  editMode,
+  height,
+  maxAccess,
+  store,
+  onApply,
+  setStore
+}) => {
+  const [valueGridData, setValueGridData] = useState([])
+  const { postRequest } = useContext(RequestsContext)
+  const [refresh, setRefresh] = useState(false)
+  const { getRequest } = useContext(RequestsContext)
+  const { recordId } = store
+
+  const [applyTrigger, setApplyTrigger] = useState(0)
+
+  const formik = useFormik({
+    initialValues: {
+      indicatorData: [
+        {
+          id: 1,
+
+          strategyId: recordId,
+          seqNo: '',
+          codeId: '',
+          name: '',
+          indicatorId: '',
+          indicatorName: ''
+        }
+      ]
+    },
+
+    validationSchema: yup.object({
+      indicatorData: yup
+        .array()
+        .of(
+          yup.object()?.shape({
+            indicatorName: yup.string().required('Indicator Name is required')
+          })
+        )
+        .required('Indicator data array is required')
+    }),
+    onSubmit: values => {
+      submitIndicators(values.indicatorData)
+    }
+  })
+
+  const submitIndicators = obj => {
+    if (obj.length === 0) {
+      return
+    }
+
+    const requests = obj.map(item => {
+      const { id, indicatorRef, indicatorName, ...itemWithoutId } = item
+      itemWithoutId.strategyId = recordId
+
+      return postRequest({
+        extension: DocumentReleaseRepository.StrategyIndicator.set,
+        record: JSON.stringify(itemWithoutId)
+      })
+    })
+
+    Promise.all(requests)
+      .then(results => {
+        toast.success('Records Edited Successfully')
+      })
+      .catch(error => {
+        toast.error('An error occurred while editing records')
+      })
+  }
+
+  const getValueGridData = recordId => {
+    getRequest({
+      extension: DocumentReleaseRepository.StrategyIndicator.qry,
+      parameters: `_strategyId=${recordId}`
+    })
+      .then(res => {
+        const gridData = res.list.map((item, index) => ({
+          id: index + 1,
+          seqNo: item.seqNo,
+          name: item.name,
+          indicatorName: item.indicatorName,
+          indicatorId: item.indicatorId,
+          codeId: item.codeId,
+          strtategyId: item.strategyId
```

Comment:
are all these necessary or can be replaces by ...rest ?

Reply:
yes we can i will change it